Assets/Scripts/Actor Components/Buff.cs

```diff
@@ -23,6 +28,22 @@ public class Buff : MonoBehaviour
 
     public bool IsBuffed { get; set; }
 
+    private void Update()
+    {
+        if (isTimerRunning)
+        {
+            if (timer > 0f)
+            {
+                timer -= Time.deltaTime;
+            }
+            else 
+            {
+                RemoveBuff();
+                isTimerRunning = false;
```

Comment:
It might be better to set `isTimerRunning` to false first. This could prevent multiple calls to `RemoveBuff` in the (admittedly unlikely) event that `Update` manages to run on the next frame again before the first `RemoveBuff` completes.